### PR TITLE
Fixed setElementDimension resets attachElements

### LINKED ISSUE #3114: setElementInterior/Dimension resets attachElements
### Describe the bug

Attached elements will reset if element changes interior/dimension. Need to re-attach.

### Steps to reproduce

attachElements()
then setElementInterior/Dimension()

### Version

Multi Theft Auto v1.6-release-21889 (all prev versions too)

### Additional context

_No response_

### Relevant log output

_No response_

### Security Policy

- [X] I have read and understood the Security Policy and this issue is not security related.

### PR DESCRIPTION
**Pull request by adem-hosni**
Fixes #3114 
when use attachElements with setElementDimension all the attached elements are reset

### CODE REVIEW

File: Client/mods/deathmatch/logic/CClientEntity.cpp [hunk 1]
```diff
@@ -614,6 +614,10 @@ void CClientEntity::SetDimension(unsigned short usDimension)
     m_usDimension = usDimension;
 
     CLuaArguments Arguments;
+    for (uint i = 0; i < GetAttachedEntityCount(); i++)
+    {
+        GetAttachedEntity(i)->SetDimension(usDimension);
+    }
```
Comment (TracerDS) [inline Client/mods/deathmatch/logic/CClientEntity.cpp:620]: Use iterators instead + `auto` or `std::size_t` instead of `uint`

**Review by CrosRoad95 — COMMENTED** [review]
lgtm

**Comment by botder** [comment]
My problem with this solution is the recursive event trigger of `onClientElementDimensionChange`.

**Comment by Proxy-99** [comment]
> My problem with this solution is the recursive event trigger of `onClientElementDimensionChange`.

@HYPER-00 can u work on this and add function like 
`GetAttachedEntity(i)->SetAttachElemenstDimension(usDimension);`

I think you should include server side too 
Server\mods\deathmatch\logic\CElement.cpp

**Review by TracerDS — CHANGES_REQUESTED** [review]

**Pull request closed without merging**